src/main/java/org/sunbird/workflow/service/impl/UserProfileWfServiceImpl.java

```diff
@@ -64,11 +68,140 @@ public void updateUserProfile(WfRequest wfRequest) {
 	}
 
 	private void updateProfile(WfRequest wfRequest) {
-		StringBuilder builder = new StringBuilder();
-		String endPoint = configuration.getHubProfileUpdateEndPoint().replace(Constants.USER_ID_VALUE,
-				wfRequest.getApplicationId());
-		builder.append(configuration.getHubServiceHost()).append(endPoint);
-		requestServiceImpl.fetchResult(builder, wfRequest.getUpdateFieldValues(), Map.class);
+		try {
+			String updatedDeptName = null;
+			String existingDeptName = null;
+			List<HashMap<String, Object>> updatedFieldValues = wfRequest.getUpdateFieldValues();
+			HashMap<String, Object> updatedFieldValueElement = updatedFieldValues.get(0);
+			HashMap<String, Object> toValueList = (HashMap<String, Object>) updatedFieldValueElement.get(Constants.TO_VALUE);
+			HashMap<String, Object> fromValueList = (HashMap<String, Object>) updatedFieldValueElement.get(Constants.FROM_VALUE);
+			for (String key : fromValueList.keySet()) {
+				if (Constants.NAME.equals(key)) {
+					existingDeptName = (String) fromValueList.get(Constants.NAME);
+				}
+			}
+			for (String key : toValueList.keySet()) {
+				if (Constants.NAME.equals(key)) {
+					updatedDeptName = (String) toValueList.get(Constants.NAME);
+				}
+			}
+			if (null != updatedDeptName) {
+					wfRequest.setDeptName(updatedDeptName);
+					Map<String, Object> response = (Map<String, Object>) migrateUser(wfRequest);
+					if (null != response && !Constants.OK.equals(response.get(Constants.RESPONSE_CODE))) {
+						logger.error("Migrate user failed" + ((Map<String, Object>) response.get(Constants.PARAMS)).get(Constants.ERROR_MESSAGE));
+						failedCase(wfRequest);
+						return;
+				}
+			}
+
```

Comment:
Are we not checking this request is for updating "professionalDetails" -- ?
what will happen if the from and to value contains "name" field -- for some other object in future ?